docs/extensions.rst

```diff
@@ -133,3 +133,38 @@ TLS
 ---
 
 See :doc:`specs/tls`.
+
+HTTP Trailers
+---
+
+The Trailer response header allows the sender to include additional fields at the
+end of chunked messages in order to supply metadata that might be dynamically
+generated while the message body is sent, such as a message integrity check,
+digital signature, or post-processing status.
+
+ASGI servers that implement this extension will provide
+``http.response.trailers`` in the extensions part of the scope::
+
+    "scope": {
+        ...
+        "extensions": {
+            "http.response.trailers": {},
+        },
+    }
+
+An ASGI framework interested in sending trailing headers to the client, must include
+the ``Trailer`` header in the ``http.response.start``. That will allow the ASGI server
+to know that after the last ``http.response.body`` message (``more_body`` being ``False``),
+the ASGI framework will send a ``http.response.trailers`` message.
```

Comment:
```suggestion
An ASGI framework interested in sending trailing headers should include the ``Trailer`` header
in the *Response Start*, as mentioned in the [RFC 2616](https://www.rfc-editor.org/rfc/rfc2616.html#section-14.40).

The ASGI framework that wants to send trailing headers can send multiple *Response Body* messages,
but the last one should set ``more_body`` as ``True``. This will tell the ASGI server to wait for
more messages, on which the ASGI framework can send multiple ``http.response.trailers`` messages.
The last ``http.response.trailers`` on its turn should set ``more_body`` as ``False``.
```

Reply:
The ```more_body```  can be ``False`` by default if not exists, just follow the one in ```http.response.body```

Reply:
So multiple ``http.response.trailers`` messages is allowed, but if you just want to send one, you may not to add ``more_body`` field. This is what ``http.response.body`` did.

Reply:
Yeah, that field is False by default, as mentioned below.

Reply:
I think that multiple `http.response.trailers` messages is an unnecessary complication - headers are unlikely to be streamed and I don't believe it makes sense to send more body after the trailers. Therefore I'd leave it for the ASGI app to figure out what trailers to send and have a single message sent to the server.

Reply:
@pgjones Then the server needs to be aware that the trailing headers are going to be present on `http.response.start`, and `more_body` is not needed on `http.response.trailers` (what is currently commited).

Then are we fine on saying that the ASGI framework MUST send the `Trailer` header in the `http.response.start`?

Reply:
I think so - does the ASGI app need to give a value equal to the headers it intends to send as well?

Reply:
> I think that multiple http.response.trailers messages is an unnecessary complication 

Actually when you are using [h11](https://pypi.org/project/h11/), there is only one chance to send trailer: the end of response body.  So it's ok for servers which only support http/1.1. But when it comes to http/2, things changed. Mutiple trailers is allowed, and, to be honest, I spend quite a lot time thinking trailers under http/2, and found that may need some [tricky](https://github.com/nonebot/nonecorn/issues/11). I have no idea if it's allowed.

Reply:
> I think so - does the ASGI app need to give a value equal to the headers it intends to send as well?

In case the ASGI app doesn't send the `Trailer` header, how does the server can be aware when the response is completed? 

On the `uvicorn` PR mentioned on the description, it uses a new field `trailers` on the `http.response.start`. On the `nonecorn` implementation, it makes use of the `more_body` on `http.response.body` and `http.response.trailers`.

Reply:
I agree with the ASGI app only needing to include a `Trailer` header, my question was if the server should check the values of that header against the trailers sent.

Reply:
Ah! I would naively say "no". As the RFC mentioned above says, the server SHOULD (and not MUST) send the `Trailer` header, I don't think it's necessary for the server to make sure the trailings headers are the same as the ones in the `Trailer` header.

Should I clarify this in the spec?